cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1221,14 +1255,18 @@ rmm::device_buffer reader::impl::decompress_page_data(
           break;
         default: CUDF_EXPECTS(false, "Unexpected decompression dispatch"); break;
       }
+      decompress_check(inflate_out.device_ptr(start_pos), argc - start_pos, stream, any_page_failure.device_ptr());
```

Comment:
When using `device_span`, you'd create a subspan similar to how we do it in the `snappy_decompress()` above.

Reply:
Actually, in this case, you don't need to check after each loop. You can check all `inflate_out` at once outside the loop. They are non-overlapping.

Reply:
Oh right that makes it easier - fixed.